docs/development/pigs/pig-21.rst

```diff
@@ -0,0 +1,348 @@
+
+
+Proposal
+========
+
+Spectral Norm Models
+--------------------
+
+For the purpose of adjusting template based models we propose to introduce a new
+class of spectral models. Those spectral models feature a norm parameter instead
+of amplitude and are named using the ``NormSpectralModel`` suffix:
+
+.. code::
+
+    from gammapy.modeling.models import (
+        PowerLawNormSpectralModel,
+        LogParabolaNormSpectralModel,
+        PiecewiseBrokenPowerlawNormSpectralModel,
+        ConstantNormSpectralModel,
+    )
+
+    pwl_norm = PowerLawNormSpectralModel(norm=, tilt=, reference=)
+    log_parabola_norm = LogParabolaNormSpectralModel(norm=, alpha=, beta=)
+    bpwl_norm = PiecewiseBrokenPowerlawNormSpectralModel(norms=, energies=)
+    const_norm = ConstantNormSpectralModel(norm=)
+
+    # typically used like
+	template = TemplateSpectralModel()
+    model = template * pwl_norm
+
+    model = BackgroundModel(
+		map=map,
+		spectral_model=pwl_norm
+	)
+
+This requires removing the hard-coded parameters of the `TemplateSpectralModel` and `BackgroundModel`.
+
+Energy Dependent Spatial Models
+-------------------------------
+A very common use-case in scientific analyses is to look for energy dependent
+morphology of extended sources. In the past this kind of analysis has been typically
+done by splitting the data into energy bands and doing individual fits of the
+morphology in these energy bands. In a combined spectral and spatial ("cube") analysis
+this can be naturally achieved by allowing for an energy dependent spatial model,
+where the energy dependence is e.g. described by a parametric model expression with
+energy dependent parameters.
+
+In the current model scheme we use a "factorised" representation of the source model,
+where the spatial, energy and time dependence are independent model components and
+not correlated:
+
+.. math::
+
+    f(l, b, E, t) = A \cdot F(l, b) \cdot G(E) \cdot H(t)
+
+To represent energy dependent morphology we propose to introduce energy
+dependent spatial models:
+
+.. math::
+
+    f(l, b, E, t) = A \cdot F(l, b, E) \cdot G(E) \cdot H(t)
+
+In general the energy dependence is optional. If the spatial model does not declare
+an energy dependence it assumes the same morphology for all energies. This also ensures backwards
+compatibility with the current behaviour.
+
+To limit the implementation effort in this PIG we propose to only adapt the ``SkyDiffuseCube``
+and add an example energy dependent custom model to our documentation. We do not propose to
+introduce general dependence of arbitrary model parameters for any spatial model, such as
+``GaussianSpatialModel`` or ``DiskSpatialModel``. An example of how this can be achieved with
+a custom implemented model is given below.
+
+We propose to add energy dependence to the ``TemplateSpatialModel`` and
+replace the current ``SkyDiffuseCube`` by:
+
+
+.. code::
+
+    spatial_model = TemplateSpatialModel.read("my_cube.fits")
+
+    model = SkyModel(
```

Comment:
I think this is really confusing. 
If a spectral model is supposed to return a flux it should be clearly doing this. So passing a `NormModel` for a `SpectralModel` is confusing.
I'd rather have the possibility to do :
```
pwl_norm = PowerLawNormModel()   (or MultModel)
template = TemplateSpatialModel.read("my_cube.fits")
template = pwl_norm * template
```
or if implementing the `CoumpoundModel` for the template is too complex having a property to set the spectral normalization (or simply passing it on init).

BTW, with this scheme (`TemplateSpatialModel` for `SkyDiffuseCube`) we have to find a way to make it clear which component actually carries the proper flux unit.

Reply:
That's the main point of this proposal: giving up the strong assumption that a `SpectralModel` should return a flux. This means we don't have a single convention, but rely on users combining the models correctly instead, if the flux is contained in the spatial model component instead.

I can see how the solution is simple and clean:
```
pwl_norm = PowerLawNormModel()   (or MultModel)
template = TemplateSpatialModel.read("my_cube.fits")
template = pwl_norm * template
```
But right now the model that combines the spectral, spatial and temporal is the `SkyModel`. So the last line in the code example `template = pwl_norm * template` should actually return a `SkyModel`...the only alternative solution would be that it returns an (energy-dependent) `SpatialModel`, but I think this would be even more confusing and require that we allow to use `SkyModel` without a spectral model, like:

```
pwl_norm = PowerLawNormModel()   (or MultModel)
template = TemplateSpatialModel.read("my_cube.fits")
template = pwl_norm * template

source = SkyModel(spatial_model=template)
```
That's why I proposed to combine it on the `SkyModel` level...